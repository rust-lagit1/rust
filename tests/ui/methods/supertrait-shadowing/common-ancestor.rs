//@ run-pass
//@ check-run-results

#![feature(supertrait_item_shadowing)]
#![allow(dead_code)]

trait A {
    fn hello(&self) {
        println!("A");
    }
}
impl<T> A for T {}

trait B: A {
    fn hello(&self) {
        println!("B");
    }
}
impl<T> B for T {}

fn main() {
    ().hello();
    //~^ WARN trait method `hello` from `B` shadows identically named method from supertrait
}
