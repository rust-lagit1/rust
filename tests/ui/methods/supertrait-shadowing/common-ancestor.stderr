warning: trait method `hello` from `B` shadows identically named method from supertrait
  --> $DIR/common-ancestor.rs:22:8
   |
LL |     ().hello();
   |        ^^^^^
   |
note: method from `B` shadows a supertrait method
  --> $DIR/common-ancestor.rs:15:5
   |
LL |     fn hello(&self) {
   |     ^^^^^^^^^^^^^^^
note: method from `A` is shadowed by a subtrait method
  --> $DIR/common-ancestor.rs:8:5
   |
LL |     fn hello(&self) {
   |     ^^^^^^^^^^^^^^^
   = note: `#[warn(supertrait_item_shadowing)]` on by default

warning: 1 warning emitted

